src/Build/BackEnd/Components/SdkResolution/SdkResolution.cs

```diff
@@ -0,0 +1,198 @@
+        /// <returns>Path to the root of the referenced SDK.</returns>
+        internal string GetSdkPath(ReferencedSdk sdk, ILoggingService logger, BuildEventContext buildEventContext,
+            ElementLocation importElementLocation, string solutionPath)
+        {
+            ErrorUtilities.VerifyThrowInternalNull(sdk, nameof(sdk));
+            ErrorUtilities.VerifyThrowInternalNull(logger, nameof(logger));
+            ErrorUtilities.VerifyThrowInternalNull(buildEventContext, nameof(buildEventContext));
+            ErrorUtilities.VerifyThrowInternalNull(importElementLocation, nameof(importElementLocation));
+
+            if (_resolvers == null) Initialize(logger, buildEventContext, importElementLocation);
+
+            var results = new List<SdkResultImpl>();
+
+            try
+            {
+                var buildEngineLogger = new SdkLoggerImpl(logger, buildEventContext);
+                foreach (var sdkResolver in _resolvers)
+                {
+                    var context = new SdkResolverContextImpl(buildEngineLogger, importElementLocation.File, solutionPath);
+                    var resultFactory = new SdkResultFactoryImpl(sdk);
+                    try
+                    {
+                        var result = (SdkResultImpl)sdkResolver.Resolve(sdk, context, resultFactory);
+                        if (result != null && result.Success)
+                        {
+                            LogWarnings(logger, buildEventContext, importElementLocation, result);
+                            return result.Path;
+                        }
+
+                        results.Add(result);
+                    }
+                    catch (Exception e)
+                    {
+                        logger.LogFatalBuildError(buildEventContext, e, new BuildEventFileInfo(importElementLocation));
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogFatalBuildError(buildEventContext, e, new BuildEventFileInfo(importElementLocation));
+                throw;
+            }
+
+            foreach (var result in results)
+            {
+                LogWarnings(logger, buildEventContext, importElementLocation, result);
+
+                if (result.Errors != null)
+                {
+                    foreach (var error in result.Errors)
+                    {
+                        logger.LogErrorFromText(buildEventContext, "", "", "",
+                            new BuildEventFileInfo(importElementLocation), error);
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private void Initialize(ILoggingService logger, BuildEventContext buildEventContext, ElementLocation location)
+        {
+            lock (_lockObject)
+            {
+                if (_resolvers != null) return;
+                _resolvers = _sdkResolverLoadingStrategy.LoadResolvers(logger, buildEventContext, location);
```

Comment:
Wonder if it's worth logging the resolver order when the log is diagnostic. Could be really helpful when debugging the resolver chain.